contracts/WETH9.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+// Copied from the verified code from Etherscan:
+// https://etherscan.io/address/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2#code
+// And then updated for Solidity version 0.6. Specific changes:
```

Comment:
```suggestion
// And then updated for Solidity version 0.8. Specific changes:
```

Reply:
That comment was actually pre-existing from the variant that I copied from the UMA repository, so I think it shold stay as-is, because the only change I made for newer versions of solidity was marking transferFrom() as virtual. This is unfortunately the downside of having copied and modified the content in the same commit :(

Reply:
Maybe I should change the comment to identify the UMA repo commit where I got it from :thinking:.

Reply:
Have updated the comment to clarify that we sourced it from UMAprotocol/protocol (w/ commit hash), and went further by restoring the proper license header. We've previously improperly removed attribution to Dapphub, and it's a requirement of the GPL that we retain the pre-existing license header.

Reply:
Nice +1